flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/OperatorCoordinatorCheckpoints.java

```diff
@@ -121,14 +123,17 @@ private static void acknowledgeAllCoordinators(
                 final Throwable error =
                         checkpoint.isDisposed() ? checkpoint.getFailureCause() : null;
 
+                CheckpointFailureReason reason = CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE;
                 if (error != null) {
-                    throw new CheckpointException(
-                            errorMessage,
-                            CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE,
-                            error);
+                    Throwable rootThrowable = ExceptionUtils.stripCompletionException(error);
```

Comment:
Again, I don't sure that we should have `stripCompletionException` here, perhaps, it should be `findThrowable`